src/overlays/actors/ovl_En_Clear_Tag/z_en_clear_tag.c

```diff
@@ -61,25 +80,993 @@ static ColliderCylinderInit D_809D5C40 = {
     },
     { 15, 30, 10, { 0, 0, 0 } },
 };
-*/
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Clear_Tag/func_809D35B0.s")
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Clear_Tag/func_809D36A8.s")
+static u32 D_809D5C98 = 0; // unused
+static u32 D_809D5C9C = 0; // unused
+
+static EnClearTagEffect clearTagParticlesBuffer[CLEAR_TAG_PARTICLE_MAX_COUNT];
+
+#include "../../../../assets/overlays/ovl_En_Clear_Tag/ovl_En_Clear_Tag.c"
+
+/**
+ * Creates a debris particle.
+ * Debris particles are spawned when the Arwing dies. It spawns fire particles.
+ */
+void EnClearTag_CreateDebrisParticle(GlobalContext* globalCtx, Vec3f* position, Vec3f* velocity, Vec3f* acceleration,
+                                     f32 scale, f32 floorHeight) {
+    s16 i;
+    s16 seed;
+    EnClearTagEffect* effect;
+    GlobalContext* globalCtx2;
+
+    globalCtx2 = globalCtx;
```

Comment:
I think it's more consistent within this function to NOT do that since moving the two variables set there to the declarations is not OK.

Reply:
I think what Zel meant was to have `globalCtx2` as the name for the argument, and `globalCtx` in the function, and I agree it would look better since we could continue using the name `globalCtx` as usual

Reply:
I disagree with that change. It's 1) Inconsistent with how other functions are defined. It's always `globalCtx` and not `globalCtx2`, and 2) That inconsistency will make a regex much more difficult when we actually implement passing in the `game` base state struct rather than `game_play` derived state struct.

Reply:
Having `globalCtx2` for the argument instead of the temp actually makes things much easier for us in the future, since we can just replace the argument to something like `gameState` instead of `globalCtx2` when we make the change. And the `globalCtx` temp, which is the one that's actually used in the rest of the function, won't have to be touched or renamed.

Reply:
heavily agreed here. when the time comes to cast gameState to globalCtx everywhere, renaming the arg is much easier than having to rename the variable used throughout the function